src/contrib/cluster/Akka.Cluster.Sharding.Tests.MultiNode/AsyncWriteProxyEx.cs

```diff
@@ -410,49 +410,60 @@ internal static IActorRefProvider ResolveProvider(ICanTell self)
             return null;
         }
 
-        private static Task<object> AskEx(ICanTell self, Func<IActorRef, object> messageFactory, IActorRefProvider provider, TimeSpan? timeout, CancellationToken cancellationToken)
+        private static async Task<object> AskEx(ICanTell self, Func<IActorRef, object> messageFactory, IActorRefProvider provider, TimeSpan? timeout, CancellationToken cancellationToken)
```

Comment:
In this method I did a few changes:
1. Setting the right exception in case of Timeout
2. With the help of the async/await - removing one level of indirection(unregister callback is gone) - this logic happens in the try/finaly
3. Right order of disposing things in the finally block